rules/DeadCode/Rector/Stmt/RemoveNextSameValueConditionRector.php

```diff
@@ -0,0 +1,139 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\DeadCode\Rector\Stmt;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr\Variable;
+use PhpParser\Node\Stmt\If_;
+use Rector\Contract\PhpParser\Node\StmtsAwareInterface;
+use Rector\DeadCode\SideEffect\SideEffectNodeDetector;
+use Rector\PhpParser\Node\BetterNodeFinder;
+use Rector\Rector\AbstractRector;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @see \Rector\Tests\DeadCode\Rector\Stmt\RemoveNextSameValueConditionRector\RemoveNextSameValueConditionRectorTest
+ */
+final class RemoveNextSameValueConditionRector extends AbstractRector
+{
+    public function __construct(
+        private readonly SideEffectNodeDetector $sideEffectNodeDetector,
+        private readonly BetterNodeFinder $betterNodeFinder,
+    ) {
+    }
+
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition(
+            'Remove already checked if condition repeated in the very next stmt',
+            [
+                new CodeSample(
+                    <<<'CODE_SAMPLE'
+final class SomeClass
+{
+    public function __construct(array $items)
+    {
+        $count = 100;
+        if ($items === []) {
+            $count = 0;
+        }
+
+        if ($items === []) {
+            return $count;
+        }
+    }
+}
+CODE_SAMPLE
+
+                    ,
+                    <<<'CODE_SAMPLE'
+final class SomeClass
+{
+    public function __construct(array $items)
+    {
+        $count = 100;
+        if ($items === []) {
+            $count = 0;
+            return $count;
+        }
+    }
+}
+CODE_SAMPLE
+                ),
+
+            ]
+        );
+    }
+
+    /**
+     * @return array<class-string<Node>>
+     */
+    public function getNodeTypes(): array
+    {
+        return [StmtsAwareInterface::class];
+    }
+
+    /**
+     * @param StmtsAwareInterface $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        if ($node->stmts === null) {
+            return null;
+        }
+
+        foreach ($node->stmts as $key => $stmt) {
+            if (! $stmt instanceof If_) {
+                continue;
+            }
+
+            // first condition must be without side effect
+            if ($this->sideEffectNodeDetector->detect($stmt->cond)) {
```

Comment:
Does this line skip property-hooks? 

A property hook might initialize a expr on read

Reply:
I am on mobile, property hooks is not in the loop of stmtsaware interface afaik.

But for property fetch, indeed need verification for reassign, even on variable, see my previous comment https://github.com/rectorphp/rector-src/pull/7525#pullrequestreview-3363582744

Also needs check to skip for common getter, eg:

```php
function get()
{
    if (! $this->prop instanceof stdClass) {
         $this->prop = $this->getStd();
    }

   if (! $this->prop instanceof stdClass) {
         echo "fetch failure";
    }
    
    return $this->prop;
}
```

This assign also need to skip, even on normal variable.